KarhooUISDK/Classes/Payment/AddCard/AdyenCardRegistrationFlow.swift

```diff
@@ -112,38 +120,83 @@ extension AdyenCardRegistrationFlow: DropInComponentDelegate {
+            guard var adyenData = Utils.convertToDictionary(data: jsonData) else {
+                return
+            }
 
-        paymentService.getAdyenPaymentDetails(paymentDetails: request).execute(callback: { result in
-            print("payment details response ", result)
+            adyenData["paymentData"] = data.paymentData
+            adyenData["details"] = adyenData
+            submitAdyenPaymentDetails(payload: adyenData)
+        } catch let error {
+            adyenDropIn?.stopLoading()
+            didFail(with: error, from: component)
+        }
+    }
+
+    private func submitAdyenPaymentDetails(payload: [String: Any]) {
+        let request = PaymentsDetailsRequestPayload(transactionID: self.transactionId,
+                                                    paymentsPayload: payload)
+
+        paymentService.getAdyenPaymentDetails(paymentDetails: request).execute(callback: { [weak self] result in
+            guard let self = self else { return }
+
+            switch result {
+            case .success(let result):
+                guard let detailsRespone = Utils.convertToDictionary(data: result.data) else {
+                    self.finish(result: .completed(value: .didFailWithError(nil)))
+                    return
+                }
+
+                let event = self.adyenResponseHandler.nextStepFor(data: detailsRespone,
+                                                                   transactionId: self.transactionId)
+                self.handle(event: event)
+            case .failure(let error):
+                self.finish(result: .completed(value: .didFailWithError(error)))
+            }
         })
     }
 
     func didFail(with error: Error, from component: DropInComponent) {
-        print("didFail", error, component)
-        adyenDropIn?.viewController.dismiss(animated: true, completion: nil)
+        if let error = error as? ComponentError, error == .cancelled {
+            finish(result: .completed(value: .cancelledByUser))
+        } else {
+            finish(result: .completed(value: .didFailWithError(error as? KarhooError)))
+        }
     }
 
     private func handle(event: AdyenResponseHandler.AdyenEvent) {
         switch event {
-        case .failure: print("payment failure!")
+        case .failure:
+            finish(result: .completed(value: .didFailWithError(nil)))
         case .paymentAuthorised(let transactionId):
-            adyenDropIn?.viewController.dismiss(animated: true, completion: nil)
             let method = PaymentMethod(nonce: transactionId)
-            callback?(.completed(value: .didAddPaymentMethod(method: method)))
+            finish(result: .completed(value: .didAddPaymentMethod(method: method)))
         case .requiresAction(let action):
             adyenDropIn?.handle(action)
-        case .handleResult(let code): print("adyen haldner result: \(code)")
+        case .handleResult(let code):
+            print("adyen haldner result: \(code)")
```

Comment:
Also this?

Reply:
https://karhoo.atlassian.net/browse/MOB-2731 once this is in the console print should be gone!